arm_controller:
    type: position_controllers/JointTrajectoryController
    joints:
        - shoulder_pan_joint
        - shoulder_lift_joint
        - elbow_pan_joint
        - elbow_lift_joint
        - wrist_pan_joint
        - wrist_lift_joint

    constraints:
        goal_time: 0.6
        stopped_velocity_tolerance: 0.05
        shoulder_pan_joint: {trajectory: 10, goal: 10}
        shoulder_lift_joint: {trajectory: 10, goal: 10}
        elbow_pan_joint: {trajectory: 10, goal: 10}
        elbow_lift_joint: {trajectory: 10, goal: 10}
        wrist_pan_joint: {trajectory: 10, goal: 10}
        wrist_lift_joint: {trajectory: 10, goal: 10}

    stop_trajectory_duration: 0.5
    state_publish_rate: 25
    action_monitor_rate: 10

gripper_controller:
    type: position_controllers/JointTrajectoryController
    joints:
        - gripper_left_joint
        - gripper_right_joint

    constraints:
        goal_time: 0.6
        # vel_tolerance is 0 for due to the movegroup/follow_joint_trajectory/goal tolerance is empty
        stopped_velocity_tolerance: 0.05
        gripper_right_joint: {trajectory: 10, goal: 10}
        gripper_left_joint: {trajectory: 10, goal: 10}

    stop_trajectory_duration: 0.5
    state_publish_rate: 25
    action_monitor_rate: 10
    gains: # Required because we're controlling an effort interface
        gripper_right_joint: {p: 100,  d: 1, i: 1, i_clamp: 1}
        gripper_left_joint: {p: 100,  d: 1, i: 1, i_clamp: 1}


head_controller:
    type: position_controllers/JointTrajectoryController
    joints:
        - joint_camera_pan
        - joint_camera_tilt

    constraints:
        goal_time: 0.6
        stopped_velocity_tolerance: 0.05
        joint_camera_pan: {trajectory: 10, goal: 10}
        joint_camera_tilt: {trajectory: 10, goal: 10}

    stop_trajectory_duration: 0.5
    state_publish_rate: 25
    action_monitor_rate: 10

base_controller:
    type: position_controllers/JointTrajectoryController
    joints:
        - joint_base_link_to_base_wheelBL
        - joint_base_link_to_base_wheelBR
        - joint_base_link_to_base_wheelFL
        - joint_base_link_to_base_wheelFR

    constraints:
        goal_time: 0.6
        stopped_velocity_tolerance: 0.05
        joint_base_link_to_base_wheelBL: {trajectory: 10, goal: 10}
        joint_base_link_to_base_wheelBR: {trajectory: 10, goal: 10}
        joint_base_link_to_base_wheelFL: {trajectory: 10, goal: 10}
        joint_base_link_to_base_wheelFR: {trajectory: 10, goal: 10}
    stop_trajectory_duration: 0.5
    state_publish_rate: 25
    action_monitor_rate: 10